s3-handler/main.go

```diff
@@ -244,6 +263,16 @@ func main() {
 	if os.Getenv("FILTER_PATTERNS") != "" {
 		filterPatterns = strings.Split(os.Getenv("FILTER_PATTERNS"), ",")
 	}
+
+	if os.Getenv("REDACT_PATTERN") != "" {
+		redactPatternString := os.Getenv("REDACT_PATTERN")
```

Comment:
We can get and check the regex once instead of twice here.
```suggestion
	if redactPatternString = os.Getenv("REDACT_PATTERN"); redactPatternString != "" {
```